src/context.rs

```diff
@@ -1121,3 +1363,231 @@ pub fn read_resource_lines(path: &str) -> io::Result<Vec<String>> {
+        let mut ctxt: Context = Context::new(
+            None, "foo-dc".to_string(), 0, RetryInterval::new("1,2").expect("REASON"),
+        );
+
+        assert!(ctxt.partition_row_presets.is_empty(), "The 'partition_row_presets' HashMap should be empty");
+        let foo_value = ctxt.partition_row_presets.get(&name_foo);
+        assert_eq!(None, foo_value);
+
+        tokio::runtime::Runtime::new().unwrap().block_on(async {
+            ctxt.init_partition_row_distribution_preset(
+                &name_foo, 1000, 10, &"100:1").await
+        }).expect(&format!("The '{}' preset must have been created successfully", name_foo));
+        assert!(!ctxt.partition_row_presets.is_empty(), "The 'partition_row_presets' HashMap should not be empty");
+        ctxt.partition_row_presets.get(&name_foo).expect(&format!("Preset with name '{}' was not found", name_foo));
+
+        let absent_bar = ctxt.partition_row_presets.get(&name_bar);
+        assert_eq!(None, absent_bar, "{}", format!("The '{}' preset was expected to be absent", name_bar));
+
+        tokio::runtime::Runtime::new().unwrap().block_on(async {
+            ctxt.init_partition_row_distribution_preset(
+                &name_bar, 1000, 10, &"90:1,10:2").await
+        }).expect(&format!("The '{}' preset must have been created successfully", name_bar));
+        ctxt.partition_row_presets.get(&name_bar).expect(&format!("Preset with name '{}' was not found", name_bar));
+    }
+
+    fn false_input_for_partition_row_distribution_preset(
+        preset_name: String,
+        row_count: u64,
+        rows_per_partitions_base: u64,
+        rows_per_partitions_groups: String,
+    ) {
+        let mut ctxt: Context = Context::new(
+            None, "foo-dc".to_string(), 0, RetryInterval::new("1,2").expect("REASON"),
+        );
+        let result = tokio::runtime::Runtime::new().unwrap().block_on(async {
+            ctxt.init_partition_row_distribution_preset(
+                &preset_name, row_count, rows_per_partitions_base, &rows_per_partitions_groups).await
+        });
+
+        assert!(matches!(result, Err(ref _e)), "Error result was expected, but got: {:?}", result);
+    }
+
+    #[test]
+    fn test_partition_row_distribution_preset_07_neg_empty_preset_name() {
+        false_input_for_partition_row_distribution_preset("".to_string(), 1000, 10, "100:1".to_string())
+    }
+
+    #[test]
+    fn test_partition_row_distribution_preset_08_neg_zero_rows() {
+        false_input_for_partition_row_distribution_preset("foo".to_string(), 0, 10, "100:1".to_string())
+    }
+
+    #[test]
+    fn test_partition_row_distribution_preset_09_neg_zero_base() {
+        false_input_for_partition_row_distribution_preset("foo".to_string(), 1000, 0, "100:1".to_string())
+    }
+
+    #[test]
+    fn test_partition_row_distribution_preset_10_neg_percentage_is_not_100() {
+        false_input_for_partition_row_distribution_preset("foo".to_string(), 1000, 10, "90:1,9:2".to_string())
```

Comment:
I think you missing the case the input is above 100%

Reply:
It is `90+9 = 99%`.

Reply:
I didn't see a case like 90+15, as a negative case I've meant 

I assume from what I've read, that it would fail, right ?

Reply:
Right, it will fail.

Reply:
Added one more test.